go/gilded-rose.go

```diff
@@ -4,31 +4,42 @@ type Item struct {
 	name            string
 	sellIn, quality int
 }
+func ReduceSulfuraConjureAndOthersQuality(item *Item) {
+	if item.quality > 0 {
+		if item.name != "Sulfuras, Hand of Ragnaros" {
+			if item.name == "Conjured Mana Cake" {
+				item.quality = item.quality - 2
+			} else {
+				item.quality = item.quality - 1
+			}
+		}
+	}
+}
+
+func ExtraBackstageOperations(item *Item){
+	if item.name == "Backstage passes to a TAFKAL80ETC concert" {
+		if item.sellIn < 11 {
+			if item.quality < 50 {
+				item.quality = item.quality + 1
+			}
+		}
+		if item.sellIn < 6 {
+			if item.quality < 50 {
+				item.quality = item.quality + 1
+			}
+		}
+	}
+}
```

Comment:
I would suggest not splitting these out unless you're going to reuse them somewhere, and/or it helps with testing. This doesn't seem to be the case though, and for me at least, this would just give me more mental overhead.

This is just a comment though, admittedly I didn't rewrite these ugly nested if's in my solution because I wasn't sure how to actually improve it without rewriting all of it, and I wanted to avoid doing that :see_no_evil: